unattended.sh

```diff
@@ -1,178 +1,615 @@
-      error_and_exit "Could not disable SELinux. You might need to run this script as root."
-    fi
-    print_step_end
-  else
-    print_step_end "OK, already disabled"
-  fi
-else
-  print_step_end "OK, not detected"
-fi
+	log "INFO" "Check if the system OS is supported and set the environment variables"
+
+  get_os_release
+
+	case "$os_version" in
+    7*)
+      log "INFO" "Setting specific part for v7 ($os_version)"
+     
+      case "$os_release" in
+        centos-release* | centos-linux-release*)
+          BASE_PACKAGES=(centos-release-scl)
+          ;;
+        
+        oraclelinux-release*|enterprise-release*)
+          BASE_PACKAGES=(oraclelinux-release-el7)
+          ;;
+      esac
+
+      RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el7/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el7.centos.noarch.rpm"
+		  PHP_BIN="/opt/rh/rh-php72/root/bin/php"
+		  PHP_ETC="/etc/opt/rh/rh-php72/php.d"
+		  OS_SPEC_SERVICES="rh-php72-php-fpm httpd24-httpd"
+		  PKG_MGR="yum"
+		  CENTREON_REPO="centreon-$repo\*"
+
+      log "INFO" "Installing required base packages"
+      if ! $PKG_MGR -y -q install ${BASE_PACKAGES[@]}
+      then
+        error_and_exit "Failed to install required base packages  ${BASE_PACKAGES[@]}"
+      fi
+		  ;;
 
+    8*) 
+      log "INFO" "Setting specific part for v8 ($os_version)"
+      
+      case "$os_release" in
+        centos-release* | centos-linux-release*)
+          BASE_PACKAGES=(dnf-plugins-core epel-release)
+          dnf config-manager --set-enabled powertools
+          ;;
+        
+        oraclelinux-release*|enterprise-release*)
+          BASE_PACKAGES=(dnf-plugins-core oracle-epel-release-el8)
+          dnf config-manager --set-enabled ol8_codeready_builder
+          ;;
+      esac
+
+      #FIXME check the result
+		  dnf -y install ${BASE_PACKAGES[@]}
+		  dnf -y update gnutls
+		 
+	  	
+      RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el8/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el8.noarch.rpm"
+		  PHP_BIN="/bin/php"
+	  	PHP_ETC="/etc/php.d"
+		  OS_SPEC_SERVICES="php-fpm httpd"
+		  PKG_MGR="dnf"
+		  CENTREON_REPO="centreon-$repo*"
+		  ;;
+	  
+	  *)
+		  error_and_exit "This '$script_short_name' script only supports Red-Hat compatible distribution (v7 and v8). Please check https://documentation.centreon.com/$CENTREON_MAJOR_VERSION/en/installation/introduction.html for alternative installation methods."
+		  ;;
+	esac
+	
+}
+#========= end of function set_required_prerequisite()
+
+
+#========= begin of function is_systemd_present()
 #
-# REPOSITORY
+function is_systemd_present(){
+	# systemd check.
+	command -v systemctl > /dev/null 2>&1
+	if [ "x$?" '=' x0 ] ; then
+	  systemctl show > /dev/null 2>&1
+	  if [ "x$?" '=' x0 ] ; then
+		has_systemd=1
+	  else
+		has_systemd=0
+	  fi
+	else
+	  has_systemd=0
+	fi
+}
+#========= end of function is_systemd_present()
```

Comment:
Could we be using this to check the OS is using systemd (or not) ?
`ps --no-headers -o comm1`